pkg/gen/python-templates/yaml.py.mustache

```diff
@@ -48,14 +51,14 @@ class ConfigFile(pulumi.ComponentResource):
         else:
             text = read_file(file_id)
 
-        # TODO: transformation support
         if opts is not None:
             _opts = deepcopy(opts)
             _opts.parent = self
         else:
             _opts = pulumi.ResourceOptions(parent=self)
 
-        self.register_outputs(parse_yaml_document(yaml.safe_load_all(text), _opts))
+        output = parse_yaml_document(yaml.safe_load_all(text), _opts, transformations)
+        self.register_outputs({"output": output})
```

Comment:
This could use a comment:

"Unlike NodeJS, Python requires that we "pull" on our futures in order to get them scheduled for execution. In order to do this, we leverage the engine's RegisterResourceOutputs to wait for the resolution of all resources that this YAML document created."